### ISSUE TITLE
Pre-build DevWorkspace templates for devfile v2 of devfile registry

### ISSUE THREAD

**Issue by benoitf** [eclipse-che/che#19718]
### Is your enhancement related to a problem? Please describe.
Creating a devWorkspace from a devfile result in dashboard may require several operations like:
- grabbing the content definition of the selected editor,
- grabbing recommended plug-ins from the projects
- grabbing plug-ins definition and potential sidecars to use
etc.

At the end, we have a workspace definition (DevWorkspace templates) that can be used to create the DevWorkspace

### Describe the solution you'd like
pre-built the devWorkspace templates for some editors like theia/intellij, etc and host them in the devfile registry
then, when dashboard wants to create a devWorkspace from a given 'getting started' devfile, it'll only need to apply the templates on k8s api instead of parsing/analyzing/etc.

**Comment by sleshchenko** [comment]
Dropping area/dashboard since it's more on devfile-registry currently.
Please return it back or even better raise a new issue when dashboard already can start handing that.

**Comment by svor** [comment]
The idea is to generate DW templates at build time and put them into 
`/usr/local/apache2/htdocs/devfiles/v2/{editor}/` folder in che-devfile-registry image.
Firstly let's do it for che-theia editor (`/usr/local/apache2/htdocs/devfiles/v2/che-theia/`). 
It could be done by using [devworkspace-handler library](https://github.com/eclipse-che/che-theia/tree/main/tools/devworkspace-handler). To use this library you can execute `node lib/entrypoint.js --devfile-url:https://github.com/benoitf/console-java-simple/tree/devfile2 --output-file:/tmp/all-ione.yaml`.
cc @vinokurig @benoitf

**Comment by max-cx** [comment]
Hi, a question to the assignee of this issue:

Will the outcome require any changes to the relevant content of the _Installation Guide_ or _Administration Guide_ or _End-user Guide_?

Yes/No?

**Comment by vinokurig** [comment]
Is blocked by https://github.com/eclipse/che/issues/20780

**Comment by vinokurig** [comment]
@max-cx
> Hi, a question to the assignee of this issue:
>
>Will the outcome require any changes to the relevant content of the Installation Guide or Administration Guide or End-user Guide?
>
>Yes/No?

I don't think so, It's just a technical debt.

**Comment by svor** [comment]
Links of DevWorkspaceTemplates should be added into `meta.yaml` at build time   
and put generated files into devfile folder. 
For example final `meta.yaml` file for Go devfile should be like:

```json
{
  "displayName": "Go",
  "description": "Stack with Go 1.14",
  "tags": [
    "Debian",
    "Go"
  ],
  "icon": "/images/go.svg",
  "links": {
    "v2": "https://github.com/che-samples/golang-echo-example/tree/devfile2",
    "devWorkspaces": {
       "eclipse/che-theia/latest": "/devfiles/go/devworkspace-che-theia-latest.yaml"
       "eclipse/che-theia/next": "/devfiles/go/devworkspace-che-theia-next.yaml"
    },
    "self": "/devfiles/go/devfile.yaml"
  }
}
```
